server/build/webpack.js

```diff
@@ -10,6 +10,12 @@ import WatchRemoveEventPlugin from './plugins/watch-remove-event-plugin'
 import DynamicEntryPlugin from './plugins/dynamic-entry-plugin'
 import DetachPlugin from './plugins/detach-plugin'
 
+// get all the environment variables
+const env = Object.keys(process.env).reduce(function (o, k) {
```

Comment:
This is not secure, it's easy for somebody to know this hack it. Specially, using third party libraries.

So, I suggest to follow what CRA does for this. [See](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/config/env.js#L15).

I think using `NEXT_` as the prefix is a good idea.